torchtext/data/utils.py

```diff
@@ -13,11 +14,57 @@ def _spacy_tokenize(x, spacy):
     return [tok.text for tok in spacy.tokenizer(x)]
 
 
+_normalize_pattern_re = [re.compile(r'\''), re.compile(r'\"'),
+                         re.compile(r'\.'), re.compile(r'<br \/>'),
+                         re.compile(r','), re.compile(r'\('),
+                         re.compile(r'\)'), re.compile(r'\!'),
+                         re.compile(r'\?'), re.compile(r'\;'),
+                         re.compile(r'\:'), re.compile(' +')]
+
+replaced_string = [' \'  ', '',
+                   ' . ', ' ',
+                   ' , ', ' ( ',
+                   ' ) ', ' ! ',
+                   ' ? ', ' ',
+                   ' ', ' ']
+
+
+def _basic_english_normalize(line):
+    r"""
+    Basic normalization for a line of text.
+    Normalization includes
+    - lowercasing
+    - complete some basic text normalization for English words as follows:
+        add spaces before and after '\''
+        remove '\"',
+        add spaces before and after '.'
+        replace '<br \/>'with single space
+        add spaces before and after ','
+        add spaces before and after '('
+        add spaces before and after ')'
+        add spaces before and after '!'
+        add spaces before and after '?'
+        replace ';' with single space
+        replace ':' with single space
+        replace multiple spaces with single space
+
+    Returns a list of tokens after splitting on whitespace.
+    """
+
+    line = line.lower()
+    for pattern_re, replaced_str in zip(_normalize_pattern_re, replaced_string):
+        line = pattern_re.sub(replaced_str, line)
+    return line.split()
+
+
 def get_tokenizer(tokenizer, language='en'):
     # default tokenizer is string.split(), added as a module function for serialization
```

Comment:
We need docs for this part (you can leave it to me) since this function will be called by users.